cmd/endpoint.go

```diff
@@ -720,18 +720,31 @@ func GetRemotePeers(endpoints EndpointList) []string {
 func updateDomainIPs(endPoints set.StringSet) {
 	ipList := set.NewStringSet()
 	for e := range endPoints {
-		host, _, err := net.SplitHostPort(e)
+		// net.SplitHostPort requires that IPv6 address are in brackets
+		if IsIPv6(e) && !strings.HasPrefix(e, "[") {
+			e = "[" + e + "]"
```

Comment:
endpoints already have `[ ]` I don't think we need to add this manually.

Reply:
I added this because if a user uses `MINIO_PUBLIC_IPS="::FFFF:0A00:0001"` the domainIPs list will contain this value without brackets (which `net.SplitHostPort()` fails on).  Similarily, if endpoints are defined via `uniqueArgs` there is nothing in the code that forces brackets onto the addresses.  

I felt that making `updateDomainIPs()` tolerant of IPv6 addresses with or without brackets was better than trying to enforce `MINIO_PUBLIC_IPS` or `uniqueArgs` to always include brackets (since these are user provided inputs).

Reply:
The problem is allowing multiple ways to do the same thing, anyone who uses IPv6 knows that they have to use `[]` to specify IPv6 address. 

We should fail appropriately and not encourage nonstandard behavior.

Reply:
I agree with your principal in general.  But I think it's worth pointing out that within Go itself functions such as `net.LookupIP` and `net.ParseIP` require that brackets _are not_ included in IPv6 addresses.  As such, I don't believe it's necessarily true that IPv6 addresses having brackets is considered standard for cases where a port is not also included.

```
net.LookupIP("::FFFF:0A00:0001") # success
net.LookupIP("[::FFFF:0A00:0001]") # error
```

It just happens to be a strange quirk of `net.SplitHostPort` that if you don't include brackets you get an error "too many colons in address" instead of "missing port in address"

```
host, port, err := net.SplitHostPort("::FFFF:0A00:0001")
# err "too many colons in address"

host, port, err := net.SplitHostPort("[::FFFF:0A00:0001]")
# err "missing port in address"
```

That all said, I don't personally care about IPv6 behavior, these were all just corner cases I observed when making unittests for `updateDomainIPs` that included IPv6 addresses.